src/main/java/se/sundsvall/eventlog/api/model/Event.java

```diff
@@ -0,0 +1,184 @@
+package se.sundsvall.eventlog.api.model;
+
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Schema;
+import org.springframework.format.annotation.DateTimeFormat;
+import se.sundsvall.dept44.common.validators.annotation.ValidUuid;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Objects;
+
+@Schema(description = "Event model")
+public class Event {
+
+	@Schema(implementation = EventType.class, enumAsRef = true)
+	@NotNull
+	private EventType eventType;
+
+	@Schema(description = "Event description", example = "Request saved in database")
+	@NotBlank
+	private String message;
+
+	@Schema(description = "Date when event can be scheduled for delete. 'null' means never", example = "2030-10-31T01:30:00.000+02:00", nullable = true)
+	@DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+	private OffsetDateTime expireDate;
+
+	@Schema(description = "Service that created event", example = "SupportManagement")
+	@NotBlank
+	private String owner;
+
+	@Schema(description = "Timestamp when the event was created", example = "2000-10-31T01:30:00.000+02:00", accessMode = Schema.AccessMode.READ_ONLY)
+	@DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+	private OffsetDateTime timestamp;
+
+	@Schema(description = "External reference for event", example = "fbe2fb67-005c-4f26-990f-1c95b5f6933e", nullable = true)
```

Comment:
The description says: "External reference for event"
The attribute is named: historyReference

Is it a requirement that this reference should point to a "historyReference" or can this be any kind of reference (decided by the calling client)? Is it necessary for eventlog-service to know what this reference is used for?

Maybe rename to "externalReference" (like the description says)?

Reply:
The idea is for other mircoservices in our ecosystem to adopt the same name for fetching a "revision" or the actual content by exposing a resource /history/{uuid}. This is meant to be a reference to an immutable snapshot of data. This would then be a convention between services made by utvecklingsfabriken and if adopted it will be very clear what this actually mean. This attribute is not really necessary since it could be added as metadata. If we however decide to adopt this naming convention, it will be a clear pattern between services that makes much more sense than "externalReference", which is used in many services with another meaning.

So, I would like to keep the attribute name, update the description to describe the purpose of the attribute, something like "Historical external reference to and immutable snapshot of data". If someone want to add a "externalReference" they are free to do so as metadata.